src/ng-select/lib/ng-select.component.ts

```diff
@@ -1,1105 +1,1134 @@
+            this.handleClearClick();
+            return;
+        }
+
+        if (target.classList.contains('ng-arrow-wrapper')) {
+            this.handleArrowClick();
+            return;
+        }
+
+        if (target.classList.contains('ng-value-icon')) {
+            return;
+        }
+
+        if (!this.focused) {
+            this.focus();
+        }
+
+        if (this.searchable) {
+            this.open();
+        } else {
+            this.toggle();
+        }
+    }
+
+    handleArrowClick() {
+        if (this.isOpen) {
+            this.close();
+        } else {
+            this.open();
+        }
+    }
+
+    handleClearClick() {
+        if (this.hasValue) {
+            this.itemsList.clearSelected(true);
+            this._updateNgModel();
+        }
+        this._clearSearch();
+        this.focus();
+        this.clearEvent.emit();
+
+        this._onSelectionChanged();
+    }
+
+    clearModel() {
+        if (!this.clearable) {
+            return;
+        }
+        this.itemsList.clearSelected();
+        this._updateNgModel();
+    }
+
+    writeValue(value: any | any[]): void {
+        this.itemsList.clearSelected();
+        this._clearSearch();
+        this._handleWriteValue(value);
+        this._setInputValue();
+        this._cd.markForCheck();
+    }
+
+    registerOnChange(fn: any): void {
+        this._onChange = fn;
+    }
+
+    registerOnTouched(fn: any): void {
+        this._onTouched = fn;
+    }
+
+    setDisabledState(state: boolean): void {
+        this._disabled = state;
+        this._cd.markForCheck();
+    }
+
+    toggle() {
+        if (!this.isOpen) {
+            this.open();
+        } else {
+            this.close();
+        }
+    }
+
+    open() {
+        if (this.disabled || this.isOpen || this._manualOpen) {
+            return;
+        }
+
+        if (!this._isTypeahead && !this.addTag && this.itemsList.noItemsToSelect) {
+            return;
+        }
+        this.isOpen = true;
+        this.itemsList.markSelectedOrDefault(this.markFirst);
+        this.openEvent.emit();
+        if (!this.searchTerm) {
+            this.focus();
+        }
+        this.detectChanges();
+    }
+
+    close() {
+        if (!this.isOpen || this._manualOpen) {
+            return;
+        }
+
+        this.isOpen = false;
+        this._isComposing = false;
+
+        if (this.multiple) {
+            this._clearSearch();
+        } else {
+            this._setInputValue();
+        }
+
+        if (!this._editableSearchTerm) {
+            this._clearSearch();
+        } else {
+            this.itemsList.resetFilteredItems();
+        }
+
+        this.itemsList.unmarkItem();
+        this._onTouched();
+        this.closeEvent.emit();
+        this._cd.markForCheck();
+    }
+
+    toggleItem(item: NgOption) {
+        if (!item || item.disabled || this.disabled) {
+            return;
+        }
+
+        if (this.deselectOnClick && item.selected) {

```

Comment:
We use `multiple` in our Multi-Select Dropdowns, does this change impact that since it replaces the `multiple` check with `deselectOnClick` which we won't be adding at this time?

Reply:
@geebru I will test this.

Reply:
@geebru I don't see any breaking changes within the ng-select demo, I will test again with CAweb after the release candidate release